libethcore/BasicAuthority.cpp

```diff
@@ -50,8 +50,12 @@ void BasicAuthority::generateSeal(BlockHeader const& _bi)
 	BlockHeader bi = _bi;
 	h256 h = bi.hash(WithoutSeal);
 	Signature s = sign(m_secret, h);
-	setSig(bi, s);
-	SealEngineBase::generateSeal(bi);
+    setSig(bi, s);
+    RLPStream ret;
+    bi.streamRLP(ret);
+    if (m_onSealGenerated)
+        m_onSealGenerated(ret.out());
+    // Else throw an error!!!
```

Comment:
I'd say it's a problem of the calling code if it would not set this handler and then wait somehow (how?) for it to be called.
So I don't think anything need to be done here.